### PR TITLE
[PIN-2354] Adapted calls using tenant-management/process

### PR DESCRIPTION
**Pull request by bloodsky**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala [hunk 1]
```diff
@@ -93,8 +94,10 @@ final case class GatewayApiServiceImpl(
   ): Route = authorize {
     logger.info(s"Upserting tenant with extenalId ($origin,$externalId) and attribute $code")
 
+    // TODO Here we need a call to Party-Registry-Proxy to retrieve institution form IPA and get the name
+
     val result: Future[Unit] = tenantProcessService
-      .upsertTenant(m2mTenantSeedFromApi(origin, externalId)(code))
+      .upsertTenant(m2mTenantSeedFromApi(origin, externalId, "Waiting for IPA")(code))
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala]: Which is the field you filled with `"Waiting for IPA"`?

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala]: NVM, is `name`, I saw it later, sorry

File: src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala [hunk 2]
```diff
@@ -610,28 +612,39 @@ final case class GatewayApiServiceImpl(
...
       id = eService.id,
-      producer = producer.toModel(tenant.id),
+      producer = tenant.toModel(category),
       name = eService.name,
       version = latestDescriptor.version,
       description = eService.description,
       technology = eService.technology.toModel,
       attributes = attributes,
       state = state
     )
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala:638]: Add a space between methods/functions/expressions. I fear that your ide might be misconfigured, check if the formatter is `scalafmt`

File: src/main/scala/it/pagopa/interop/apigateway/api/impl/package.scala [hunk 3]
```diff
@@ -271,23 +271,20 @@ package object impl extends SprayJsonSupport with DefaultJsonProtocol {
     def toModel: EServiceDoc = EServiceDoc(id = doc.id, name = doc.name, contentType = doc.contentType)
   }
 
-  implicit class EnrichedInstitution(private val institution: PartyManagementApiInstitution) extends AnyVal {
-    def toModel(tenantId: UUID): Organization =
+  implicit class EnrichedTenant(private val tenant: Tenant) extends AnyVal {
+    def toModel(category: String): Organization = {
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/apigateway/api/impl/package.scala]: Useless parenthesis

File: src/main/scala/it/pagopa/interop/apigateway/server/impl/Dependencies.scala [hunk 4]
```diff
@@ -146,12 +136,11 @@ trait Dependencies {
...
   def gatewayApi(jwtReader: JWTReader, blockingEc: ExecutionContextExecutor)(implicit
     actorSystem: ActorSystem[_],
-    ec: ExecutionContext,
-    partyManagementApiKeyValue: PartyManagementApiKeyValue
+    ec: ExecutionContext
   ): GatewayApi =
     new GatewayApi(
       GatewayApiServiceImpl(
-        partyManagementService(),
+        // partyManagementService(),
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/apigateway/server/impl/Dependencies.scala]: 🧹

**Review by TonioGela — CHANGES_REQUESTED** [review]
LGTM, but I asked for a few changes

File: src/main/scala/it/pagopa/interop/apigateway/service/impl/PartyRegistryProxyServiceImpl.scala [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    extends PartyRegistryProxyService {
+  implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  override def getInstitutionByExternalId(origin: String, originId: String)(implicit
+    contexts: Seq[(String, String)]
+  ): Future[Institution] = for {
+    (bearerToken, correlationId, ip) <- extractHeaders(contexts).toFuture
+    request = api.getInstitutionByExternalId(
+      // xCorrelationId = correlationId,
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/apigateway/service/impl/PartyRegistryProxyServiceImpl.scala]: 🧹

File: src/main/scala/it/pagopa/interop/apigateway/service/impl/PartyRegistryProxyServiceImpl.scala [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      origin = origin,
+      originId = originId
+      // xForwardedFor = ip
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/apigateway/service/impl/PartyRegistryProxyServiceImpl.scala]: 🧹

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/apigateway/service/impl/PartyRegistryProxyServiceImpl.scala (hunk 5 above)
Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/apigateway/service/impl/PartyRegistryProxyServiceImpl.scala]: Nope, once [this](https://github.com/pagopa/interop-be-party-registry-proxy/pull/144) gets merged you need to add these params

**Review by beetlecrunch — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**